app/model/APIDefinition.scala

```diff
@@ -0,0 +1,99 @@
+/*
+ * Copyright 2016 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package model
+
+import model.APIStatus.APIStatus
+import model.CollaboratorRole.CollaboratorRole
+import play.api.libs.json.Json
+
+case class APISubscription(name: String, serviceName: String, context: String, versions: Seq[VersionSubscription], requiresTrust: Option[Boolean])
+
+case class APIDefinition(
+                          serviceName: String,
+                          serviceBaseUrl: String,
+                          name: String,
+                          description: String,
+                          context: String,
+                          versions: Seq[APIVersion],
+                          requiresTrust: Option[Boolean]) {
+
+  require(serviceName.nonEmpty, s"serviceName is required")
+  require(serviceBaseUrl.nonEmpty, s"serviceBaseUrl is required")
+  require(name.nonEmpty, s"name is required")
+  require(context.nonEmpty, s"context is required")
+  require(description.nonEmpty, s"description is required")
+  require(versions.nonEmpty, s"at least one version is required")
+  require(uniqueVersions, s"version numbers must be unique")
```

Comment:
Probably not necessary to enforce requirements about data that's owned by a different service?